### PR TITLE
Orphan DaemonSet when deleting with --cascade option set

### PR DESCRIPTION
**Pull request by soltysh**
**What this PR does / why we need it**:
This is a followup to https://github.com/kubernetes/kubernetes/pull/64847 to ensure that `--cascade` options is properly handled for DaemonSet

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
This was originally described in https://bugzilla.redhat.com/show_bug.cgi?id=1623352

For review:
/assign @mfojtik 
For 1.11 approval:
/assign @foxish 

cc @janetkuo 

**Release note**:

```release-note
Orphan DaemonSet when deleting with --cascade option set
```

### CODE REVIEW

File: pkg/kubectl/cmd/delete.go [hunk 1]
```diff
@@ -321,8 +321,9 @@ func (o *DeleteOptions) deleteResource(info *resource.Info, deleteOptions *metav
 	// TODO: Remove this in or after 1.12 release.
 	//       Server version >= 1.11 no longer needs this hack.
 	mapping := info.ResourceMapping()
-	if mapping.Resource.GroupResource() == (schema.GroupResource{Group: "extensions", Resource: "daemonsets"}) ||
-		mapping.Resource.GroupResource() == (schema.GroupResource{Group: "apps", Resource: "daemonsets"}) {
+	if (mapping.Resource.GroupResource() == (schema.GroupResource{Group: "extensions", Resource: "daemonsets"}) ||
+		mapping.Resource.GroupResource() == (schema.GroupResource{Group: "apps", Resource: "daemonsets"})) &&
+		(*deleteOptions.PropagationPolicy == metav1.DeletePropagationOrphan) {
```
Comment (tnozicka) [inline pkg/kubectl/cmd/delete.go]: shouldn't it be `!=` possibly with checking the nil value of delete options?

Comment (soltysh) [reply pkg/kubectl/cmd/delete.go]: It never will be nil, we're setting it to `metav1.DeletePropagationBackground` always a couple of lines before calling `deleteResource` and only conditionally setting `metav1.DeletePropagationOrphan`.

Comment (tnozicka) [reply pkg/kubectl/cmd/delete.go]: ok, but the hack (forcing the pods to be deleted) should only be called if the policy **is not** DeletePropagationOrphan, right?

Comment (tnozicka) [reply pkg/kubectl/cmd/delete.go]: /hold

**Comment by soltysh** [comment]
Fixed the error @tnozicka pointed out.

/hold cancel

**Comment by soltysh** [comment]
Re-applying the lgtm label back.

**Comment by soltysh** [comment]
@foxish for the 1.11 approval

**Pull request merged**